app/controllers/contacts_controller.rb

```diff
@@ -61,16 +72,18 @@ def set_contact
   end
 
   def set_new_contact_from_params
-    @contact = Contact.new(contact_params)
+    @contact = Contact.last.update!(contact_params)
```

Comment:
You can't rely on `last` giving you the expected Contact because 2 users could be creating an address at the same time.